include/onnxruntime/core/platform/threadpool.h

```diff
@@ -172,13 +188,30 @@ class ThreadPool {
     }
     tp->ParallelFor(total, cost_per_unit, fn);
   }
+#endif
 
   // Similar to ParallelFor above, but takes the specified scheduling strategy
   // into account.
   void
   ParallelFor(std::ptrdiff_t total, const SchedulingParams& scheduling_params,
               const std::function<void(std::ptrdiff_t, std::ptrdiff_t)>& fn);
 
+#ifdef _OPENMP
+  static void TryParallelFor(concurrency::ThreadPool* tp, std::ptrdiff_t total, const SchedulingParams&,
+                             const std::function<void(std::ptrdiff_t, std::ptrdiff_t)>& fn) {
+    std::ptrdiff_t num_threads = concurrency::ThreadPool::NumThreads(tp);
```

Comment:
nit: slightly more readable if there are names for the fn params like

                   const std::function<void(std::ptrdiff_t first, std::ptrdiff_t last)>& fn);